addons/mail/static/src/components/thread_viewer/thread_viewer_tests.js

```diff
@@ -476,6 +476,70 @@ QUnit.test('show message subject if thread is mailing channel', async function (
     )
 });
 
+QUnit.test('show only messages that need to be shown', async function (assert) {
```

Comment:
I'm not sure if you intend to be complete with this test, but there are other possible types (email, and then also sms and snailmail).

Maybe it would be enough to limit the current test to ensuring `user_notification` are not shown. Other tests in here already implicitly ensure other types are visible.

Reply:
difficult to test that.. indeed the message is not stored in the store at all as the 'RPC' does not return it... So you can't determine the local id of the message that should not be printed...